gems/sorbet-runtime/lib/types/private/methods/inline_type_validation.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+# typed: false
+
+module T::Private::Methods::InlineTypeValidation
+  def self.report_error(error, value, type: nil, location: nil)
+    # Backwards compatibility before `inline_type_error_handler` took a second arg
+    if T::Configuration.inline_type_error_handler.arity == 1
+      return T::Configuration.inline_type_error_handler(e)
+    end
```

Comment:
This doesn't seem quite right: this expression would be testing the arity of `T::Configuration.inline_type_error_handler` itself or maybe the value returned by a zero-argument send to `T::Configuration.inline_type_error_handler` (?).  But what you want to be checking is the arity of the thing that would be called by `T::Configuration.inline_type_error_handler` (`@inline_type_error_handler` or `inline_type_error_handler_default`)...right?

I guess you could just dispense with this helper module and move the arity logic into `T::Configuration.inline_type_error_handler`?  That would change the public interface of `T::Configuration`, which you might have been trying to avoid.  I'm not sure what the right thing to do here is.

Reply:
Ah I saw the `@inline_type_error_handler` instance variable and assumed it was an `attr_accessor`. Moving this handling into `T::Configuration.inline_type_error_handler` definitely seems like the right move. I'll update this and add some tests.